### PR TITLE
Add Idea Hub events for pagination (3980).

### PR DESCRIPTION
**Pull request by johnPhillips**
## Summary

Addresses issue #3980

## Relevant technical choices
I deviated from the IB slightly, as when I started implementing this it felt sensible to pull the tracking calls out into a function. 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js [hunk 1]
```diff
@@ -60,19 +62,47 @@ const Pagination = ( { tab } ) => {
...
-		}
-	}, [ page, setValue, uniqueKey ] );
+	const handlePrev = useCallback(
+		( shouldTrackEvent = true ) => {
+			if ( page > 1 ) {
+				setValue( uniqueKey, page - 1 );
+			}
+			if ( shouldTrackEvent ) {
+				trackPage( 'back' );
+			}
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js:95]: We should call the `trackPage` function only if the current page allows the user to go back. In other words, if the current page is greater than `1`.

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js [hunk 2]
```diff
@@ -60,19 +62,47 @@ const Pagination = ( { tab } ) => {
...
+		},
+		[ page, setValue, uniqueKey, trackPage ]
+	);
 
 	const handleNext = useCallback( () => {
 		if ( page < Math.ceil( total / IDEA_HUB_IDEAS_PER_PAGE ) ) {
 			setValue( uniqueKey, page + 1 );
 		}
-	}, [ page, setValue, total, uniqueKey ] );
+		trackPage( 'forward' );
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js:104]: Same here. We need to make sure that we track events when the page is actually changed.

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js [hunk 3]
```diff
@@ -60,19 +62,47 @@ const Pagination = ( { tab } ) => {
...
+				},
+				back: {
+					'new-ideas': 'new_page_return',
+					'save-ideas': 'saved_page_return',
+					'draft-ideas': 'draft_page_return',
+				},
+			};
+			const event = eventMap[ direction ][ tab ];
+
+			trackEvent( IDEA_HUB_GA_CATEGORY_WIDGET, event, page );
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js:81]: Let's probably add a check to make sure that `event` is not empty.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @johnPhillips. Mostly looks good to me. Added a few minor comments and it should be good to go.

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js (hunk 1 above)
Comment (johnPhillips) [reply assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js:95]: Great catch - this should be in the `if` block above.

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js (hunk 3 above)
Comment (johnPhillips) [reply assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Pagination.js:81]: Good idea

**Review by eugene-manuilov — APPROVED** [review]
LGTM

**Pull request merged**